src/autofix/autofix.ts

```diff
@@ -0,0 +1,304 @@
+	// Only collect and parse files for which fixes are available
+	const resources: Resource[] = [];
+	for (const [_, autofixResource] of autofixResources) {
+		const messagesById = getAutofixMessages(autofixResource);
+		// Currently only global access autofixes are supported
+		// This needs to stay aligned with the applyFixes function
+		if (messagesById.has(MESSAGE.NO_GLOBALS)) {
+			resources.push(autofixResource.resource);
+		}
+	}
+
+	const sourceFiles: SourceFiles = new Map();
+	const resourcePaths = [];
+	for (const resource of resources) {
+		const resourcePath = resource.getPath();
+		const sourceFile = ts.createSourceFile(
+			resource.getPath(),
+			await resource.getString(),
+			{
+				languageVersion: ts.ScriptTarget.ES2022,
+				jsDocParsingMode: ts.JSDocParsingMode.ParseNone,
+			}
+		);
+		sourceFiles.set(resourcePath, sourceFile);
+		resourcePaths.push(resourcePath);
+	}
+
+	const compilerHost = createCompilerHost(sourceFiles);
+	const program = createProgram(resourcePaths, compilerHost);
+
+	const checker = program.getTypeChecker();
+	const res: AutofixResult = new Map();
+	for (const [resourcePath, sourceFile] of sourceFiles) {
+		log.verbose(`Applying autofixes to ${resourcePath}`);
+		const newContent = applyFixes(checker, sourceFile, resourcePath, autofixResources.get(resourcePath)!);
+		if (newContent) {
+			const jsErrors = getJsErrors(newContent, resourcePath);
+			if (jsErrors.length) {
+				const message = `Syntax error after applying autofix for '${resourcePath}': ` +
+					jsErrors.map((d) => d.messageText as string).join(", ");
+				log.verbose(message);
+				log.verbose(resourcePath + ":\n" + newContent);
+				context.addLintingMessage(resourcePath, MESSAGE.PARSING_ERROR, {message});
+			} else {
+				log.verbose(`Autofix applied to ${resourcePath}`);
+				res.set(resourcePath, newContent);
+			}
+		}
+	}
+
+	return res;
+}
+
+function applyFixes(
+	checker: ts.TypeChecker, sourceFile: ts.SourceFile, resourcePath: ResourcePath,
+	resource: AutofixResource
+): string | undefined {
+	const content = sourceFile.getFullText();
+
+	// Collect modules of which at least one message has the conditional fixHint flag set
+	const conditionalModuleAccess = new Set<string>();
+	for (const msg of resource.messages) {
+		if (msg.fixHints?.moduleName && msg.fixHints?.conditional) {
+			log.verbose(`Skipping fixes that would import module '${msg.fixHints.moduleName}' ` +
+				`because of conditional global access within the current file.`);
+			conditionalModuleAccess.add(msg.fixHints.moduleName);
+		}
+	}
+
+	// Group messages by id
+	const messagesById = new Map<MESSAGE, RawLintMessage[]>();
+	for (const msg of resource.messages) {
+		if (msg.fixHints?.moduleName && conditionalModuleAccess.has(msg.fixHints.moduleName)) {
+			// Skip messages with conditional fixHints
+			continue;
+		}
+		if (!messagesById.has(msg.id)) {
+			messagesById.set(msg.id, []);
+		}
+		messagesById.get(msg.id)!.push(msg);
+	}
```

Comment:
Isn't this the same as `getAutofixMessages()`?

Reply:
Oh yeah, I think I missed to also refactor this part. Thanks.

Reply:
Done